lib/models/blueprint.js

```diff
@@ -1027,12 +1027,14 @@ let Blueprint = CoreObject.extend({
   addBowerPackagesToProject(packages, installOptions) {
     let task = this.taskFor('bower-install');
     let installText = (packages.length > 1) ? 'install bower packages' : 'install bower package';
-    let packageNames = [];
-    let packageNamesAndVersions = packages.map(pkg => {
+
+    let packageNames = new Array(packages.length);
+    let packageNamesAndVersions = new Array(packages.length);
+    packages.forEach((pkg, i) => {
       pkg.source = pkg.source || pkg.name;
-      packageNames.push(pkg.name);
-      return pkg;
-    }).map(bowEpParser.compose);
+      packageNames[i] = pkg.name;
+      packageNamesAndVersions[i] = bowEpParser.compose(pkg);
+    });
```

Comment:
goal was just cleanup, it was using `[].push` inside map (which looks weird) also just tried avoid extra iterations,  ^ what you suggested and what I did looks equal in clarity IMHO, should redo it like you suggested still ?